### ISSUE TITLE
Should `map_clone` trigger on copies of double references?

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust-clippy#3958]
```rust
fn main() {
    let x = Some(&"abcd");
    let y = x.map(|&p| p);
}
```

[playpen](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3523fc85920de82312f67d17823e2d1f)

```
warning: You are using an explicit closure for cloning elements
 --> src/main.rs:3:13
  |
3 |     let y = x.map(|&p| p);
  |             ^^^^^^^^^^^^^ help: Consider calling the dedicated `cloned` method: `x.cloned()`
  |
  = note: #[warn(clippy::map_clone)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
```

So `map_clone` triggers even for copies, which makes sense, since `cloned()` works on copies. Here it specificially is triggering on copies of a double reference.

However, we also have [`clone_double_ref`](https://rust-lang.github.io/rust-clippy/master/index.html#clone_double_ref) which warns about cloning `&&T` since that may not do what one expected.

Also `.map(|&x| x)` is much clearer than `.cloned()` in such cases, it's not at all obvious that the `.cloned()` is there to unwrap a reference

I feel that this is a false positive, but I can see others disagreeing. Thoughts?

h/t @hawkw

**Comment by oli-obk** [comment]
Note that there's also [`.copied()`](https://doc.rust-lang.org/std/?search=fn%3A%20copied) which would make it much clearer that there's not heavy duty code happening.

**Comment by Manishearth** [comment]
In that case we should definitely suggest that whenever possible.

**Comment by Manishearth** [comment]
Fixed in https://github.com/rust-lang/rust-clippy/pull/3970, but we can't land that until `.copied()` stabilizes.

**Comment by Centril** [comment]
`.copied` was stabilized.

**Comment by Manishearth** [comment]
Sweet, put next steps here: https://github.com/rust-lang/rust-clippy/pull/4043
